contrib/mac/framework/julia-config.jl

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env julia
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+const options =
+[
+    "--cflags",
+    "--ldflags",
+    "--ldlibs"
+];
```

Comment:
In `base/Makefile`, where `build_h.jl` is generated, I added 1 or 2 variables to base:

1. `Base.DARWIN_FRAMEWORK` (boolean: true if packaged as framework, else false)
2. `Base.DARWIN_FRAMEWORK_NAME` (string: allows packager to rename the framework e.g. JuliaGPL.framework or JuliaDebug.framework).

In my limited testing I have not had issues with packages that assume things about the julia installation, but it seems very likely that some patches do indeed make assumptions.   For example, assuming the traditional Unix hierarchy structure (bin,lib,share, etc.) instead of a framework structure.  For example, within Julia itself, there are assumptions that the shared library is named `libjulia[-debug].*`, whereas for my framework target the shared library takes the name contained in the string `Base.DARWIN_FRAMEWORK_NAME`. (see changes to `test/libdl.jl`)

I see the `DARWIN_FRAMEWORK` and `DARWIN_FRAMEWORK_NAME` variables as similar in purpose to the `SYSCONFDIR`, `DATAROOTDIR`, `DOCDIR`, `LIBDIR`, and `INCLUDEDIR` variables.  Package authors should already be using those variables as necessary.